aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_api_helpers.py

```diff
@@ -4,11 +4,73 @@
 )
 from .. import unit_test_utils
 from unittest.mock import MagicMock
+from botocore.exceptions import ClientError
 
 NUMBER_OF_ITEMS_PER_PAGE = 2
 MAX_PAGES = 10
 
 
+def test_retry_not_found_exceptions_decorator_happy_case():
+    mock_afd_client = unit_test_utils.create_mock_afd_client()
+    mock_afd_client.get_labels = MagicMock()
+    mock_afd_client.exceptions.ResourceNotFoundException = ClientError
+    mock_afd_client.get_labels.side_effect = [
+        ClientError({'Code': '', 'Message': ''}, 'get_labels'),
+        {
+            'labels': [{'property': f'{j}'} for j in range(NUMBER_OF_ITEMS_PER_PAGE)]
+        }
+    ]
+
+    @api_helpers.retry_not_found_exceptions
+    def call_get_labels(frauddetector_client):
+        return frauddetector_client.get_labels()
+
+    response = call_get_labels(mock_afd_client)
+    assert len(response['labels']) == NUMBER_OF_ITEMS_PER_PAGE
+    assert mock_afd_client.get_labels.call_count == 2
```

Comment:
Would like to understand how this call count turns out to be 2.

Reply:
If you look at `mock_afd_client.get_labels.side_effect`, we are first returning a `ResourceNotFoundException` from `get_labels`. Since `call_get_labels` is marked with `@api_helpers.retry_not_found_exceptions`, the function is retried after sleeping for 1 second. The second call succeeds, so we get a success response.

Reply:
Ok, I'm following, after taking a closer look at `side_effect`. First call triggers ClientError, 2nd is a successful retrieval.